tests/interpreters/test_onnx_interpreter.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+
+from xdsl.dialects import onnx
+from xdsl.dialects.builtin import ModuleOp, TensorType, f32
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.shaped_array import ShapedArray
+from xdsl.utils.test_value import TestSSAValue
+
+try:
+    import numpy as np  # pyright: ignore
+except (ImportError, ModuleNotFoundError):
+
+    np = None
+    pytest.importorskip("numpy", reason="numpy is an optional dependency in xDSL")
+
+# load in case numpy is available
+if np is not None:
+    interpreter = Interpreter(ModuleOp([]))
+    from xdsl.interpreters.onnx import OnnxFunctions
+
+    interpreter.register_implementations(OnnxFunctions())
+
+    def test_onnx_add():
+        op = onnx.Add(
+            TestSSAValue(TensorType(f32, [2, 3])),
+            TestSSAValue(TensorType(f32, [2, 3])),
+            res_type=TensorType(f32, [2, 3]),
+        )
+
+        a = ShapedArray(np.array([1, 2, 3, 4, 5, 6]), [2, 3])  # pyright: ignore
+        b = ShapedArray(np.array([1, 4, 2, 5, 3, 6]), [2, 3])  # pyright: ignore
+        c = ShapedArray(np.array([0, 0, 0, 0, 0, 0]), [2, 3])  # pyright: ignore
+
+        c = interpreter.run_op(op, (a, b, c))  # pyright: ignore
+        assert len(c) == 1
+        assert np.array_equal(  # pyright: ignore
+            c[0].data[0][0], np.array([2, 6, 5, 9, 8, 12])  # pyright: ignore
+        )  # pyright: ignore
+        assert c[0].shape == [2, 3]
+
+    def test_onnx_sub():
+        op = onnx.Sub(
+            TestSSAValue(TensorType(f32, [2, 3])),
+            TestSSAValue(TensorType(f32, [2, 3])),
+            res_type=TensorType(f32, [2, 3]),
+        )
+
+        a = ShapedArray(np.array([1, 2, 3, 4, 5, 6]), [2, 3])  # pyright: ignore
+        b = ShapedArray(np.array([1, 4, 2, 5, 3, 6]), [2, 3])  # pyright: ignore
+        c = ShapedArray(np.array([0, 0, 0, 0, 0, 0]), [2, 3])  # pyright: ignore
+
+        c = interpreter.run_op(op, (a, b, c))  # pyright: ignore
+        assert len(c) == 1
+        assert np.array_equal(  # pyright: ignore
+            c[0].data[0][0], np.array([0, -2, 1, -1, 2, 0])  # pyright: ignore
+        )  # pyright: ignore
+        assert c[0].shape == [2, 3]
+
+    def test_onnx_mul():
+        op = onnx.Mul(
+            TestSSAValue(TensorType(f32, [2, 2])),
+            TestSSAValue(TensorType(f32, [2, 2])),
+            res_type=TensorType(f32, [2, 2]),
+        )
+
+        a = ShapedArray(np.array([1, 4, 7, 1]), [2, 2])  # pyright: ignore
```

Comment:
What's the type error? We want to avoid all ignores possible

Reply:
To have the pyright CI be aware of numpy we need to modify the CI script, just cherry pick the pyright-ci change from this PR: #2334